app/build.gradle

```diff
@@ -191,9 +192,29 @@ dependencies {
   implementation project(":domain")
   implementation project(":utility")
 }
+
 // The GeneratedMessageLite implementations of protobufs are depending on protobuf-java
 // instead of protobuf-lite after Android Studio 3.5,
 // The below command stops that from happening: https://github.com/google/tink/issues/282
 configurations {
   all*.exclude module: 'protobuf-java'
 }
+
+
+tasks.withType(Test) {
+  jacoco.includeNoLocationClasses = true
+}
+task jacocoTestReport(type: JacocoReport, dependsOn: ['testDebugUnitTest', 'createDebugCoverageReport']) {
+
+  reports {
+    xml.enabled = true
+    html.enabled = true
+  }
+
+  def mainSrc = "$project.projectDir/src/main/kotlin"
+
+  sourceDirectories.setFrom(files([mainSrc]))
+  executionData.setFrom(fileTree(dir: project.buildDir, includes: [
+      'jacoco/testDebugUnitTest.exec', 'outputs/code_coverage/debugAndroidTest/connected/**/*.ec'
```

Comment:
I think  `outputs/code_coverage/debugAndroidTest/connected/**/*.ec`  is not needed here as we only want to cover robolectric tests. Removing this makes the Gradle task work fine.

Reply:
Also if we keep this i think we will have to downgrade the version of Gradle because Gradle has an ongoing issue which mentions that since version 6.4  Gradle does not wait for the coverage report to be generated.

Reply:
[here](https://github.com/gradle/gradle/issues/14132) is the link to gradle's issue.

Reply:
Seen